packages/editor/src/components/propertyTab/properties/floatPropertyTabComponent.tsx

```diff
@@ -1,30 +1,141 @@
-import * as react from "react";
+import { Component } from "react";
+import type { ConnectionPointType, InputBlock, InputBlockEditorData } from "@babylonjs/smart-filters";
+import type { IPropertyComponentProps } from "@babylonjs/shared-ui-components/nodeGraphSystem/interfaces/propertyComponentProps.js";
 import { FloatLineComponent } from "@babylonjs/shared-ui-components/lines/floatLineComponent.js";
-import type { GlobalState } from "../../../globalState";
-import type { InputBlock } from "@babylonjs/core/Materials/Node/Blocks/Input/inputBlock";
+import { SliderLineComponent } from "@babylonjs/shared-ui-components/lines/sliderLineComponent.js";
+import { getFloatInputBlockEditorData } from "../../../graphSystem/getEditorData.js";
 
-interface IFloatPropertyTabComponentProps {
-    globalState: GlobalState;
-    inputBlock: InputBlock;
+export interface FloatPropertyTabComponentProps extends IPropertyComponentProps {
+    inputBlock: InputBlock<ConnectionPointType.Float>;
 }
 
-export class FloatPropertyTabComponent extends react.Component<IFloatPropertyTabComponentProps> {
+type FloatPropertyTabComponentState = {
+    useTime: boolean;
+    showSlider: boolean;
+};
+
+/**
+ * The property tab component for InputBlock of type ConnectionPointType.Float.
+ * If the animation type is time, display the value and valueDeltaPerMs properties
+ * plainly-- no slider capability. Otherwise, display the value alongside
+ * optional min and max properties that, when set, toggle a slider for the value.
+ */
+export class FloatPropertyTabComponent extends Component<
+    FloatPropertyTabComponentProps,
+    FloatPropertyTabComponentState
+> {
+    private _editorData: InputBlockEditorData<ConnectionPointType.Float>;
```

Comment:
Maybe I'm thinking about functional components, but is it safe to have state outside of the component's state?  I think all state needs to be kept in the react state so react can guarantee consistency.  I also think it has to be updated with .setState().

Reply:
I can remove it as a property here-- I think I had done this for readability to avoid repeating "this.props.inputBlock.editorData!".